modules/tensor_mechanics/include/materials/ComputeStrainBase.h

```diff
@@ -20,18 +20,17 @@ class ComputeStrainBase : public DerivativeMaterialInterface<Material>
 {
 public:
   ComputeStrainBase(const std:: string & name, InputParameters parameters);
+  virtual ~ComputeStrainBase() {}
 
 protected:
   virtual void initQpStatefulProperties();
   virtual void computeProperties() = 0;
 
-  VariableGradient & _grad_disp_x;
-  VariableGradient & _grad_disp_y;
-  VariableGradient & _grad_disp_z;
-
-  VariableGradient & _grad_disp_x_old;
-  VariableGradient & _grad_disp_y_old;
-  VariableGradient & _grad_disp_z_old;
+  /// Coupled displacement variables
+  unsigned int _ndisp;
+  std::vector<VariableValue *> _disp;
```

Comment:
The variable value is used in inherited classes, e.g. the Axisymmetric cases.  I put the VariableValue _disp here so that I could manage both the grad_disp and disp initialization in the same loop.  Good enough reason or would you prefer that I moved the disp VariableValue down to the Axisymmetric case?